File: PadariaDoManoel/Dominio/Entities/Produto.cs
using System;


namespace PadariaDoManoel.Entities
{
    class Produto
    {
        private int idProduto;
        private string descricao;
        private double quantidade;
        private int codigoBarras;
        private double quantidadeMinima;
        private Categoria categoria;
        private UnidadeMedida unidadeMedida;

        public Produto() { }

        public Produto(int idProduto, string descricao, double quantidade, 
            int codigoBarras, double quantidadeMinima, 
            Categoria categoria, UnidadeMedida unidadeMedida)
        {
            this.IdProduto = idProduto;
            this.Descricao = descricao;
            this.Quantidade = quantidade;
            this.CodigoBarras = codigoBarras;
            this.QuantidadeMinima = quantidadeMinima;
            this.Categoria = categoria;
            this.UnidadeMedida = unidadeMedida;
        }

        public int IdProduto { get => idProduto; set => idProduto = value; }
        public string Descricao { get => descricao; set => descricao = value; }
        public double Quantidade { get => quantidade; set => quantidade = value; }
        public int CodigoBarras { get => codigoBarras; set => codigoBarras = value; }
        public double QuantidadeMinima { get => quantidadeMinima; set => quantidadeMinima = value; }
        internal Categoria Categoria { get => categoria; set => categoria = value; }
        internal UnidadeMedida UnidadeMedida { get => unidadeMedida; set => unidadeMedida = value; }
    }
}

File: PadariaDoManoel/src/main/java/Classes/Produto.java
package Classes;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

/**
 *
 * @author Gilson
 */
@Entity
@Table (name = "TB_Produtos")


public class Produto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column (name = "IDProduto")
    private int idProduto;
    @Column (name = "Descricao")
    private String descricao;
    // Arrumar
    @Column (name = "IDUnidadeMedida")
    private int idUnidadeMedida;
    @Column (name = "IDCategoria")
    private int idCategoria;
    @Column (name = "CodigoDeBarras")
    private String codigoDeBarras;
    @Column (name = "Quantidade")
    private float quantidade;
    @Column (name = "QuantidadeMinima")
    private float quantidadeMinima;
    @Column (name = "ValorUnitario")
    private float valorUnitario;

    public Produto() {
    }

    public Produto(int idProduto, String descricao, int idUnidadeMedida, int idCategoria, String codigoDeBarras, 
            float quantidade, float quantidadeMinima, float valorUnitario) {
        this.idProduto = idProduto;
        this.descricao = descricao;
        this.idUnidadeMedida = idUnidadeMedida;
        this.idCategoria = idCategoria;
        this.codigoDeBarras = codigoDeBarras;
        this.quantidade = quantidade;
        this.quantidadeMinima = quantidadeMinima;
        this.valorUnitario = valorUnitario;
    }
    
    
    
    
    
    
}
